Added support for bulk deletes with test

agearhart opened the pull request:
Please let me know if you have any issues with this code.  :)

Commit:
Added support for bulk deletes with test

rcoh commented on elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient.scala:
```diff
@@ -248,6 +253,7 @@ object RestlasticSearchClient {
     case class BulkItem(_index: String, _type: String, _id: String, status: Int, error: Option[String]) {
       def created = status > 200 && status < 299 && !alreadyExists
       def alreadyExists = error.exists(_.contains("DocumentAlreadyExists"))
+      def deleted = status == 200
```
This is a confusing name because the same object is returned by the create action. Can you rename this to `success` and have it be >= 200 <= 299?

rcoh commented on elasticsearch-core/src/test/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClientTest.scala:
```diff
@@ -168,6 +168,52 @@ class RestlasticSearchClientTest extends WordSpec with Matchers with ScalaFuture
       }
     }
+    
+    "Support bulk deletion" in {
```
Either add deletion to the previous test, or do the setup without checking the result to avoid duplicate tests.

rcoh reviewed (changes requested):
Just a couple fixes

CCheSumo commented on elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/RestlasticSearchClient.scala:
```diff
@@ -135,6 +135,11 @@ class RestlasticSearchClient(endpointProvider: EndpointProvider, signer: Option[
     bulkIndex(bulkOperation)
   }
+  
+  def bulkDelete(index: Index, tpe: Type, documents: Seq[Document]): Future[Seq[BulkItem]] = {
```
Shouldn't we support delete by document ids instead of requiring `documents: Seq[Document]` to be passed in?

agearhart replied:
We can totally do that.  I was just keeping with conventions.  I'll prop that change soon.

Commit:
Addressing code review feedback
Renaming BulkItem.deleted to BulkItem.success
Combining bulk indexing test with bulk delete

agearhart reviewed (commented):
I've renamed BulkItem.deleted to BulkItem.success and combined the tests as suggested.  Please let me know if this is what was intended.

maryammzadeh commented:
This is awesome! I was hoping there would be support for bulk deletes soon! Thanks!

CCheSumo reviewed (commented):
@pnzrdrgoon one more comment
